debian/postinst

```diff
@@ -67,11 +68,36 @@ upgrade_to_status_cache() {
     ua status 2>&1 > /dev/null
 }
 
+is_release_upgrade() {
+    # Return 0 if the package version we upgrade from does not match the
+    # current series (e.g. 14.04)
+    case $PREVIOUS_PKG_VER in
+        *${VERSION_ID}*)  # os-release didn't change from PREVIOUS_PKG_VER
+            return 1;;
+        * ) return 0;; #  os-release changed from PREVIOUS_PKG_VER
+    esac
+}
+
 case "$1" in
     configure)
+      PREVIOUS_PKG_VER=$2
+      # Special case: legacy precise creds allowed for trusty esm
+      # do-release-upgrade substitutes s/precise/trusty/ in all apt sources.
+      # So all we need to do is rename the precise sources file to trusty.
+      # https://github.com/CanonicalLtd/ubuntu-advantage-client/issues/693
+      if is_release_upgrade; then
```

Comment:
OK, now I've seen them refactored (and possibly with fresh eyes this morning), I'm wondering if we really need this check at all. `$PREVIOUS_PKG_VER` matching `*12.04*` is a more specific condition than `$PREVIOUS_PKG_VER` _not_ matching the current release, so I think we can just drop the if statement with no change in behaviour.

What do you think?

Reply:
Sounds good @OddBloke, let's drop it for now, we'll have to re-introduce it in the future I bet with Xenial/Bionic release upgrade support.... But we'll be older and wiser developers then ;)